src/docbuilder/rustwide_builder.rs

```diff
@@ -344,16 +343,44 @@ impl RustwideBuilder {
                             &metadata,
                         )?;
                     }
-                    let new_algs = self.upload_docs(name, version, local_storage.path())?;
-                    algs.extend(new_algs);
+
+                    if self.config.use_archive_storage {
+                        let (_, new_alg) = add_path_into_remote_archive(
+                            &self.storage,
+                            &format!("rustdoc/{0}/rustdoc-{0}-{1}.zip", name, version),
```

Comment:
Hmm, why all this duplicated info?
```suggestion
                            &format!("rustdoc/{0}/{1}.zip", name, version),
```

Reply:
I think my initial idea was that the archives would be self-contained, but I'm happy to change the naming.